lib/3scale/client/http_client.rb

```diff
@@ -87,21 +90,21 @@ def post(path, payload)
         end
 
         def full_uri(path)
-          URI.join "#{@protocol}://#{@host}", path
+          URI.join "#{@protocol}://#{@host}:#{@port}", path
         end
       end
 
       class NetHttp < BaseClient
         extend Forwardable
         def_delegators :@http, :use_ssl?, :active?
 
-        def initialize(host)
+        def initialize(host, port)
           super
-          @http = Net::HTTP.new(@host, 80)
+          @http = Net::HTTP.new(@host, port)
         end
 
-        def ssl!
-          @http = Net::HTTP.new(@host, 443)
+        def ssl!(port)
+          @http = Net::HTTP.new(@host, port)
```

Comment:
Maybe here we can also get rid of this reinstantiation of `@http`?
We should already have the correct port at this point, so we only need to set `use_ssl` to true.